janusgraph-core/src/main/java/org/janusgraph/core/JanusGraphFactory.java

```diff
@@ -104,7 +116,60 @@ public static JanusGraph open(BasicConfiguration configuration) {
      * @return JanusGraph graph database
      */
     public static JanusGraph open(ReadConfiguration configuration) {
-        return new StandardJanusGraph(new GraphDatabaseConfiguration(configuration));
+        return open(configuration, null);
+    }
+
+    public static JanusGraph open(ReadConfiguration configuration, String backupName) {
+        final ModifiableConfiguration config = new ModifiableConfiguration(ROOT_NS, (WriteConfiguration) configuration, BasicConfiguration.Restriction.NONE);
+        final String graphName = config.has(GRAPH_NAME) ? config.get(GRAPH_NAME) : backupName;
+        if (null != graphName) {
+            JanusGraphManager jgm = JanusGraphManager.getInstance();
+            Preconditions.checkState(jgm != null, JANUS_GRAPH_MANAGER_EXPECTED_STATE_MSG);
+            return (JanusGraph) jgm.openGraph(graphName, (gName) -> {
+                return new StandardJanusGraph(new GraphDatabaseConfiguration(configuration));
+            });
+        } else {
+            log.info("You should supply \"graph.graphname\" in your .properties file configuration if you are opening " +
```

Comment:
Thanks. I understand now that we are talking about the "non-gremlin-server" case.

> Or are you saying that accessing graphs directly without going through GremlinServer is an unusual access pattern and should be deprecated?

No.

> In this case jgm would be expected to be null and it would be appropriate for graph.graphname to be unset, right?

I need to think more about this. My general concern is that `jgm != null` is not only an indicator of not using the gremlinServer but also of not deploying the GremlinServer using the JanusGraphManager, and I originally was thinking that would be something we want deprecated, but now I am wondering if that is valid and how it ties into this specific case here.